pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -1812,3 +1812,25 @@ CREATE TABLE duplicate_index_test (k INT PRIMARY KEY, v INT, INDEX idx (v));
 
 statement error pgcode 42P07 duplicate index name: \"idx\"
 ALTER TABLE duplicate_index_test ADD CONSTRAINT idx UNIQUE (v)
+
+##
+subtest reference_constraint_different_order
+
+# https://github.com/cockroachdb/cockroach/issues/63324
+# Referencing an existing constraint by fields in a different order
+# - should still find existing constraint
+
+statement ok
+create table reference_constraint_different_order_parent (a int not null, b int not null, c int not null, d varchar(10));
```

Comment:
nit: remove trailing `;`. we can also one line the setup to be a bit more clear:

```
statement ok
create table reference_constraint_different_order_parent (a int not null, b int not null, c int not null, d varchar(10));
create table reference_constraint_different_order_child (id int, a int, b int, c int, e varchar(10));
alter table reference_constraint_different_order_parent add constraint p_pk primary key (a, b, c)

statement ok
alter table reference_constraint_different_order_child add constraint p_fk foreign key (a, b, c) references reference_constraint_different_order_parent (a, c, b)


```

Then can you add a test of the FK succeeding / failing?

Reply:
Have matched the format you've suggested

Reply:
i would delete it from here